src/main/java/uk/gov/pay/webhooks/queue/QueueMessageReceiver.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.pay.webhooks.queue;
+
+import com.google.inject.Inject;
+import io.dropwizard.lifecycle.Managed;
+import io.dropwizard.setup.Environment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.webhooks.app.QueueMessageReceiverConfig;
+import uk.gov.pay.webhooks.app.WebhooksConfig;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class QueueMessageReceiver implements Managed {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(QueueMessageReceiver.class);
+
+    private static final String QUEUE_MESSAGE_RECEIVER_THREAD_NAME = "queue-message-receiver-%d";
+
+    private final QueueMessageReceiverConfig config;
+    private final ScheduledExecutorService scheduledExecutorService;
+    private final EventMessageHandler eventMessageHandler;
+    private final int queueReadScheduleNumberOfThreads;
+
+    @Inject
+    public QueueMessageReceiver(
+            Environment environment,
+            WebhooksConfig configuration,
+            EventMessageHandler eventMessageHandler) {
+        this.eventMessageHandler = eventMessageHandler;
+        this.config = configuration.getQueueMessageReceiverConfig();
+        this.queueReadScheduleNumberOfThreads = config.getNumberOfThreads();
+
+        scheduledExecutorService = environment
+                .lifecycle()
+                .scheduledExecutorService(QUEUE_MESSAGE_RECEIVER_THREAD_NAME)
+                .threads(queueReadScheduleNumberOfThreads)
+                .build();
+    }
+
+    @Override
+    public void start() {
+        long initialDelay = config.getThreadDelayInMilliseconds();
+        long delay = config.getThreadDelayInMilliseconds();
+
+        for(int i = 0; i < queueReadScheduleNumberOfThreads; i++) {
```

Comment:
could we do this as a forEach instead?

Reply:
Not really. `queueReadScheduleNumberOfThreads` is just an `int`. There’s no collection we’re iterating over here.